src/generator.ts

```diff
@@ -82,6 +86,9 @@ function declareNamedInterfaces(
     case 'TUPLE':
     case 'UNION':
       type = ast.params.map(_ => declareNamedInterfaces(_, options, rootASTName, processed)).filter(Boolean).join('\n')
+      if ('spreadParam' in ast && ast.spreadParam) {
```

Comment:
Nit: The second check seems to be redundant, since you defined the member type to be non-nullable.

```diff
-if ('spreadParam' in ast && ast.spreadParam) {
+if ('spreadParam' in ast) {
```

Reply:
Don't confuse TS with flow!

in TS w/ strict mode
`spreadParam?: AST` is the same as
`spreadParam?: AST | undefined`

Reply:
That's embarrassing, good catch :) Would `if (ast.spreadParam) {` work?